### PR TITLE
Add OpenAPI doc of the current API

### PR DESCRIPTION
**Pull request by paddatrapper**
This is an attempt to document the current API. The functions I have found to be broken are marked as such in the description of the function

### CODE REVIEW

File: docs/api/openapi.yaml [hunk 1]
```diff
@@ -0,0 +1,963 @@
...
+                "nextwednesday": [],
+                "nextthursday": [],
+                "nextfriday": [],
+                "nextsaturday": [],
+                "nextsunday": [],
+                "AIRTIME_API_VERSION": "1.1"
+              }
+  /station-metadata:
+    get:
+      summary: BROKEN: Retrieve the schedule for the week
```
Comment (jooola) [inline docs/api/openapi.yaml]: This string need to be quoted, it breaks when trying to parse it with swagger-ui or redoc.
Using `:` in the string can lead to weird behavior.

There might be some other strings that need to be fixed.

Comment (paddatrapper) [reply docs/api/openapi.yaml]: Thanks. Fixed

**Comment by paddatrapper** [comment]
Seems many API endpoints are broken due to setting `$this->view->msg` presumably as the response, which doesn't work. Just returns a 200 response with no content.

This is fixed by replacing that with `$this->_helper->json->sendJson(...)`. I'll open a separate PR for that

**Comment by paddatrapper** [comment]
This should now reflect the current state of the API. There are places where I am not sure what the current behaviour is, but I have tried to make them clear

**Comment by Robbt** [comment]
I'm going to commit this now since it doesn't actually affect the codebase yet and it should be useful.

**Comment by paddatrapper** [comment]
Well it is the current API, not a potential new one, so it reflects the current codebase. New API doc will follow once $dayjob settles down in the new year

**Comment by Robbt** [comment]
Cool that makes sense, should we also include the REST modules under airtime_mvc/applications/modules/rest/controllers - this is another API style functionality that it would be good to document I think.

**Comment by paddatrapper** [comment]
We should. I didn't know they existed

**Comment by Robbt** [comment]
Yeah it's part of the fragmented codebase that makes working on LibreTime so challenging. Merging these two different API points conceptually and coming up with a logical path forward would be really rewarding I suspect.

**Pull request merged**